doc/dev/ldmatrix_stmatrix.md

```diff
@@ -0,0 +1,477 @@
+
+```
+stmatrix.sync.aligned.shape.num{.trans}{.ss}.type [p], r;
+
+.shape  = {.m8n8, .m16n8};
+.num    = {.x1, .x2, .x4};
+.ss     = {.shared{::cta}};
+.type   = {.b16, .b8};
+```
+
+Reference: [StMatrix PTX](https://docs.nvidia.com/cuda/parallel-thread-execution/#warp-level-matrix-store-instruction-stmatrix)
+
+## General Details
+
+For 16-bit element size, the matrix shape is (8, 8).
+The instruction can process one, two, or four (8, 8) matrices per instruction.
+
+### Indices shared memory tensor
+Each thread in the warp specifies a matrix row in shared memory. For LdMatrix,
+the shared memory tensor is an input TensorView. For StMatrix, it is an output
+TensorView.
+
+* Threads 0-7 correspond with matrix rows of first matrix. (x1, x2, and x4)
+* Threads 8-15 correspond with matrix rows of second matrix. (x2 and x4)
+* Threads 16-23 correspond with matrix rows of third matrix. (x4 only)
+* Threads 24-31 correspond with matrix rows of fourth matrix. (x4 only)
+
+### Indices for register tensor
+For an (8, 8) matrix of 16-bit elements, this is the register layout for a warp.
+Each threads stores two adjacent elements along the inner-most dimension.
+* `[m(8), n(8)]` // An (8, 8) matrix of 16-bit elements
+* `[m(8), no(4), ni(2)]`  // Split column dimension by 2
+* `[m(8) * no(4) (TDX), ni[2])` // Merge row dimension and column stride
+
+### Register layout for one 8x8 Matrix with 16-bit elements
+![Register-Layout](https://docs.nvidia.com/cuda/parallel-thread-execution/_images/mma-stmatrix-fragments.png)
+
+# Example 1: A copy kernel using TMA, LdMatrix, and StMatrix.
+
+The first example is a simple copy kernel using only TMA, LdMatrix, and
+Stmatrix. The next example incorporates wgmma and pointwise computation into a
+fusion.
+
+The goal for this cuda kernel is to load data from shared memory, produced by
+TMA Load with 128B swizzle, to registers using LdMatrix. After some computation,
+the data is loaded from registers into shared memory, which is consumed by a
+TMA Store with 128B swizzle.
+
+## How to compute the index into register TensorView?
+
+The register index is based on loop domain and is handled by IdModel
+indexing. For LdMatrix, we must set the allocation domain using the
+scheduled loop domain.
+
+## How to compute the index into shared memory TensorView?
+
+The index into shared memory requires a custom index from loop domain to the
+TMA LoadStoreOp allocation domain.
+
+Figure 1 shows how the CTA tile is transformed into the loop domain
+for LdMatrix or StMatrix. Figure 2 displays the loop transformations for TMA
+Load or Store operations. Figure 3 illustrates the following steps to compute
+index from LdMatrix or StMatrix loop domain to TMA shared memory domain.
+
+---
+
+### Figure 1: Loop domain for LdMatrix and StMatrix
+![Register layout for LdMatrix / StMatrix](ldstmatrix/ldstmatrix_register_layout.svg)
+
+---
+
+### Figure 2: TMA shared memory domain
```

Comment:
Does `shared memory domain` refers to the allocation domain of the smem tv?

Reply:
Yes, `shared memory domain` refers to the allocation domain. 

For `scheduleTMAStoreForMmaOutput` and `scheduleTMALoadForMma`, the allocation domain is set from loop domain. Since they are the same, I didn't bother to disambiguate them.

Reply:
I updated the doc to be more clear.